EosPayload/drivers/efield_driver.py

```diff
@@ -0,0 +1,87 @@
+import logging
+import traceback
+
+
+from EosLib.format.formats.e_field import EField
+from EosLib.format import Type
+from EosLib.packet import Packet
+from EosLib.packet.data_header import DataHeader
+from EosLib.packet.definitions import Priority
+
+from EosPayload.lib.base_drivers.driver_base import DriverBase
+from EosPayload.lib.mqtt import Topic
+
+try:
+    import Adafruit_BBIO.ADC as ADC
+except ModuleNotFoundError:
+    print("ADC not installed correctly")
+    pass
+
+
+class ElectricFieldSensor(DriverBase):
+    def __init__(self, output_directory: str, config: dict):
+        super().__init__(output_directory, config)
+        self.pin_1 = "P9_36"
+        self.pin_2 = "P9_38"
+        self.pin_3 = "P9_40"
+
+    def setup(self) -> None:
+        super().setup()
+        try:
+            ADC
+        except NameError:
+            raise Exception("failed to import ADC library")
+
+        self.register_thread('device-read', self.device_read)
+
+        ADC.setup(self.pin_1)
+        ADC.setup(self.pin_2)
+        ADC.setup(self.pin_3)
+
+    def device_read(self, logger: logging.Logger) -> None:
+        adc_pins = [self.pin_1, self.pin_2, self.pin_3]
+        voltages = [None, None, None]
+        while True:
+            try:
+                # Read the voltage from the ADC pin
+                voltages = [ADC.read(pin) for pin in adc_pins]
+
+            except Exception as e:
+                self._logger.info(f"An error occurred while reading voltages: {e}\n{traceback.format_exc()}")
+
+            try:
+                row = []
+                for value in voltages:
+                    row.append(str(value))
+                # logged as Voltage 1, Voltage 2, Voltage 3
+                self.data_log(row)
+            except Exception as e:
+                logger.error(f"An unhandled exception occurred while logging data: {e}\n{traceback.format_exc()}")
+
+            try:
+                efield_obj = EField(voltages[0], voltages[1], voltages[2])
+            except Exception as e:
+                logger.error(f"exception occurred while creating EField format: {e}\n{traceback.format_exc()}")
+                self.thread_sleep(logger, 2)
+                continue
+
+            header = DataHeader(
+                data_type=Type.E_FIELD,
+                sender=self.get_device_id(),
+                priority=Priority.DATA
+            )
+            packet = Packet(
+                body=efield_obj,
+                data_header=header,
+            )
+            self._mqtt.send(Topic.RADIO_TRANSMIT, packet)
+
+            self.thread_sleep(logger, 2)
```

Comment:
do you want to read data and transmit data at the same rate?  How rapidly are the values expected to change for this sensor?  If it changes more rapidly, you may want to read and data_log every 1s or 0.5s.  How valuable is a real-time measurement at the ground station?  If its something that could suddenly spike, maybe we want to transmit every 1s.  if it's something that is going to be relatively stable, then maybe we dont need so many points on the live graph and you transmit every 5s.  Maybe log+transmit every 2s is what works for this.  Just something for you to think about